source/postprocess/velocity_points_residual.cc

```diff
@@ -0,0 +1,330 @@
+                             "located when ASPECT was compiled. This interpretation allows, "
+                             "for example, to reference files located in the `data/' subdirectory "
+                             "of ASPECT.");
+          prm.declare_entry ("Data file name", "point_values_velocity.txt",
+                             Patterns::Anything (),
+                             "The file name of the input ascii velocity data. "
+                             "The file is provided in a structured data format in one dimension "
+                             "such that the first column represents point ids from zero to n-1 "
+                             " (for n number of points) and the subsequent columns are data coordinates "
+                             "and the observed velocities at those points.");
+          prm.declare_entry ("Use spherical unit vectors", "false",
+                             Patterns::Bool (),
+                             "Specify velocity as r, phi, and theta components "
+                             "instead of x, y, and z. Positive velocities point up, east, "
+                             "and north (in 3d) or out and clockwise (in 2d). "
+                             "This setting only makes sense for spherical geometries.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    VelocityPointsResidual<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Velocity points residual");
+        {
+          // Get the path to the data files. If it contains a reference
+          // to $ASPECT_SOURCE_DIR, replace it by what CMake has given us
+          // as a #define
+          data_directory = Utilities::expand_ASPECT_SOURCE_DIR(prm.get ("Data directory"));
+          data_file_name    = prm.get ("Data file name");
+
+          use_spherical_unit_vectors = prm.get_bool("Use spherical unit vectors");
+
+          if (use_spherical_unit_vectors)
+            AssertThrow (this->get_geometry_model().natural_coordinate_system() == Utilities::Coordinates::spherical,
+                         ExcMessage ("Spherical unit vectors should not be used "
+                                     "when geometry model is not spherical."));
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(VelocityPointsResidual,
+                                  "velocity points residual",
+                                  "A postprocessor that computes some statistics about "
+                                  "the velocity residual at specified evaulation points in the input data. "
+                                  "The velocity residual is the difference between the model solution "
```

Comment:
It is the sum of the residuals at the given set of points, right? Or do you do l2 as in sqrt(sum(\|u(x_i)-uref(x_i)\|^2))? Maybe include the latex formula here?